src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -149,6 +153,35 @@ def check_lazy(*args: Any, **kwargs: Any) -> Any:
 
         return check_lazy
 
+    def __getattr__(self, key: str) -> Any:
+        """
+        Return item identified by `key`.
+
+        Parameters
+        ----------
+        key : str
+            Key to get.
+
+        Returns
+        -------
+        Any
+
+        Notes
+        -----
+        This method also helps raise NotImplementedError for APIs out
+        of current scope that are not implemented.
+        """
+        try:
+            return object.__getattribute__(self, key)
+        except AttributeError as err:
+            if not key.startswith("_"):
+                native_index = native_pd.Index([])
+                if hasattr(native_index, key):
```

Comment:
@sfc-gh-vbudati can you add a quick comment here that "any methods that not supported by the current Index.py but exists in native pandas index object should raise a not implemented error for now"

Reply:
ah I missed this - I'll add it in the documentation PR here: https://github.com/snowflakedb/snowpark-python/pull/1751

Reply:
Thanks!